cdap-app-fabric/src/main/java/co/cask/cdap/app/store/Store.java

```diff
@@ -332,29 +332,29 @@ void addApplication(Id.Application id,
   List<NamespaceMeta> listNamespaces();
 
   /**
-   * Adds adapter spec to the store. Will overwrite the existing spec.
+   * Adds adapter meta to the store. Will overwrite the existing AdapterMeta.
    *
    * @param id Namespace id
-   * @param adapterSpecification specification of the adapter
+   * @param adapterMeta metadata (including specification) of the adapter
    */
-  void addAdapter(Id.Namespace id, AdapterSpecification adapterSpecification);
+  void addAdapter(Id.Namespace id, AdapterMeta adapterMeta);
 
   /**
    * Fetch the adapter identified by the name in a give namespace.
    *
    * @param id  Namespace id.
    * @param name Adapter name
-   * @return an instance of {@link AdapterSpecification}.
+   * @return an instance of {@link AdapterMeta}.
    */
-  AdapterSpecification getAdapter(Id.Namespace id, String name);
+  AdapterMeta getAdapter(Id.Namespace id, String name);
 
   /**
    * Fetch all the adapters in a given namespace.
    *
    * @param id Namespace id.
-   * @return {@link Collection} of Adapter Specification.
+   * @return {@link Collection} of Adapter Meta.
    */
-  Collection<AdapterSpecification> getAllAdapters(Id.Namespace id);
+  Collection<AdapterMeta> getAllAdapters(Id.Namespace id);
```

Comment:
We should not change this API. ```getAllAdapters``` should return the specification and not the runtime state. All our other APIs donot mix specification with runtime state. I would recommend adding a ```status``` API to ```AdapterService``` instead to retrieve this information